### DISCUSSION TITLE
Disable indent for certain filetypes

### DISCUSSION THREAD

**Discussion by EricDriussi** [folke/snacks.nvim#371] [Q&A]
When viewing a markdown file, the indent line is always shown in the line under the current one if empty.

![image](https://github.com/user-attachments/assets/0fff1d94-ed17-4ce8-8f3e-6f769ff0bb8f)

This doesn't seem to do much for me, so I tried disabling it for that filetype:

```lua
{
    "folke/snacks.nvim",
    opts = {
        indent = {
            filter = function(buf)
                return vim.bo[buf].filetype ~= "markdown"
            end,
        },
    },
},
```

This however doesn't change the behavior.

Am I missing something? Is this a bug?

**Comment by Varal7** [comment]
The code you posted works for me. 

```
{
    "folke/snacks.nvim",
    opts = {
        indent = {
            filter = function(buf)
                return vim.bo[buf].filetype ~= "markdown"
            end,
        },
    },
},
```

However, it somehow enables indent in other buffers that don't have indent guides by default (which-key, fzf-lua, etc.)

**Reply by EricDriussi** [reply]
It does work, I had an error in my config which I did not notice before posting this discussion (big facepalm moment).
I do no t see the behavior you are describing but I'm guessing that applying this setting overrides the [default filter function](https://github.com/folke/snacks.nvim/blob/d312053f78b4fb55523def179ac502438dd93193/lua/snacks/indent.lua#L83), so  returning `vim.g.snacks_indent ~= false and vim.b[buf].snacks_indent ~= false and vim.bo[buf].buftype == "" and vim.bo[buf].filetype ~= "markdown"` instead of `vim.bo[buf].filetype ~= "markdown"` might preserve the original behavior instead of overriding it?

**Reply by Varal7** [reply]
Oh thank you! Silly me for not having checked the default value for the `filter` function.

**Comment by zeevenn** [comment]
Use mini.indentscope instead. It handle markdown very well.
